cmds/dhclient/dhclient.go

```diff
@@ -30,33 +32,48 @@ import (
 )
 
 const (
-	defaultIfname = "eth0"
 	// slop is the slop in our lease time.
-	slop = 10 * time.Second
+	slop          = 10 * time.Second
+	linkUpAttempt = 30
 )
 
 var (
+	ifName       = "^e.*"
 	leasetimeout = flag.Int("timeout", 15, "Lease timeout in seconds")
 	retry        = flag.Int("retry", -1, "Max number of attempts for DHCP clients to send requests. -1 means infinity")
-	renewals     = flag.Int("renewals", -1, "Number of DHCP renewals before exiting")
-	verbose      = flag.Bool("verbose", true, "Verbose output")
+	renewals     = flag.Int("renewals", -1, "Number of DHCP renewals before exiting. -1 means infinity")
+	verbose      = flag.Bool("verbose", false, "Verbose output")
 	ipv4         = flag.Bool("ipv4", false, "use IPV4")
 	test         = flag.Bool("test", false, "Test mode")
 	debug        = func(string, ...interface{}) {}
 )
 
 func ifup(ifname string) (netlink.Link, error) {
-	iface, err := netlink.LinkByName(ifname)
-	if err != nil {
-		return nil, fmt.Errorf("cannot get interface by name %v: %v", ifname, err)
-	}
-
-	if err := netlink.LinkSetUp(iface); err != nil {
-		return nil, fmt.Errorf("%v: %v can't make it up: %v", ifname, iface, err)
-	}
+	debug("Try bringing up %v", ifname)
+	start := time.Now()
+	for int(time.Since(start).Seconds()) < linkUpAttempt {
```

Comment:
make this

for time.Since(start) < linkUpAttempt

and 

const linkUpAttempt = 30 * time.Second